src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2302,34 +2307,91 @@ protected override void HandleSetBreakpointsRequestAsync(IRequestResponder<SetBr
 
         protected override void HandleSetExceptionBreakpointsRequestAsync(IRequestResponder<SetExceptionBreakpointsArguments> responder)
         {
-            List<string> filter = responder.Arguments.Filters;
-            if (m_engineConfiguration.ExceptionSettings.Categories.Count > 0)
+            List<ExceptionFilterOptions> filterOptions = responder.Arguments.FilterOptions;
+            if (filterOptions != null && filterOptions.Count > 0)
             {
-                if (filter == null || filter.Count == 0)
+                foreach (ExceptionFilterOptions filterOption in filterOptions)
                 {
-                    SetAllExceptions(enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE);
-                }
-                else if (filter.Contains(ExceptionBreakpointFilter.Filter_All))
-                {
-                    enum_EXCEPTION_STATE state = enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_FIRST_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_FIRST_CHANCE;
+                    if (filterOption.FilterId == ExceptionBreakpointFilter.Filter_All)
+                    {
+                        if (!string.IsNullOrWhiteSpace(filterOption.Condition))
+                        {
+                            string[] conditions = filterOption.Condition.Split(',');
```

Comment:
>conditions [](http://example.com/codeflow?start=37&length=10)

Do we want to validate the name contains valid C++ identifier customers (in case people try to write something like `== Foo` instead of `Foo`)?

Reply:
VS Code supports returning failures back for these

Reply:
Looks like VSCodeDebugAdapterHost has not been updated to return the BreskpointList.

I'll mark this in an issue for TODO.